### PR TITLE
Feat/new identities

### PR DESCRIPTION
**Pull request by JMounier**

### CODE REVIEW

**Comment by jeremad** [comment]
Please update the README (provisional + new upgrade_identity)

File: spec/tanker/identity_spec.rb [hunk 1]
```diff
@@ -181,7 +181,7 @@
       expect(identity['private_encryption_key']).to eq('tVTS9nHxr2MdVuTR5clww0EXbwis8HixgPINbQJx1U4=')
     end
 
-    it 'a valid sms public provisional identity' do
+    it 'parses a valid sms public provisional identity' do
```
Comment (theodelrieu) [inline spec/tanker/identity_spec.rb]: sms!

File: spec/tanker/identity_spec.rb [hunk 2]
```diff
@@ -128,15 +146,25 @@
       expect(identity['value']).to eq(@hashed_user_id)
     end
 
-    it 'a valid public provisional identity' do
-      identity = Tanker::Identity.deserialize(@public_provisional_identity)
+    it 'a valid non-hashed public provisional identity' do
```
Comment (theodelrieu) [inline spec/tanker/identity_spec.rb]: nitpick: unhashed or non-hashed, (nit)pick one

File: lib/tanker/identity.rb [hunk 3]
```diff
@@ -118,7 +118,7 @@ def self.get_public_identity(serialized_identity)
       public_keys.each { |key| public_identity[key] = identity.fetch(key) }
 
       if identity['target'] == 'email'
-        identity['target'] = 'hashed_email'
+        public_identity['target'] = 'hashed_email'
```
Comment (theodelrieu) [inline lib/tanker/identity.rb:119]: can fixup this commit?

File: spec/tanker/identity_spec.rb [hunk 4]
```diff
@@ -166,6 +168,32 @@
...
+      expect(identity['public_encryption_key']).to eq('n6m9X5Lf0Znaz8f0+schLIBNm+piPhnsYvAvXw2KEAw=')
+      expect(identity['private_encryption_key']).to eq('tVTS9nHxr2MdVuTR5clww0EXbwis8HixgPINbQJx1U4=')
+    end
+
+    it 'a valid sms public provisional identity' do
+      private_identity = Tanker::Identity.deserialize(@phone_number_provisional_identity)
+      identity = Tanker::Identity.deserialize(@phone_number_public_provisional_identity)
+      expect(Tanker::Identity.get_public_identity(@phone_number_provisional_identity)).to eq(@phone_number_public_provisional_identity)
+
+      hashed_phone = Tanker::Crypto.hashed_provisional_value(@user_phone, private_identity['private_signature_key'])
```
Comment (theodelrieu) [inline spec/tanker/identity_spec.rb]: nitpick: `hashed_phone_number`

**Comment by theodelrieu** [comment]
Feels like some commits could be squashed

**Review by theodelrieu — APPROVED** [review]

File: lib/tanker/identity.rb (hunk 3 above)
Comment (tux3) [reply lib/tanker/identity.rb:119]: Sure, done

**Comment by tux3** [comment]
> Feels like some commits could be squashed

That's probably because they were originally split in multiple PRs, and this is just the target branch where it got merged

**Review by blastrock — APPROVED** [review]

**Pull request merged**